### PR TITLE
Facebook authentication

### PR DESCRIPTION
**Pull request by rodrigouroz**
Implements facebook authentication and JWT tokens.

Flow:

1. The app authenticates with facebook. Gets a token.
2. The app authenticates with the server, sends the facebook token.
3. The server validates the token through facebook, if valid it issues a JWT.
4. The app then authenticates every request to the server with this JWT, until it expires.

### CODE REVIEW

**Comment by gonrial** [comment]
I've just added several fixes and a very primitive and basic errorHandler so the tests can be executed properly.
Be aware that the `errorHandler` can be split for checking the jwt auth part, the facebook auth part and as a final catchAll for all the possible 500 errors

File: server/.eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-    "extends": "airbnb-base"
+    "extends": "airbnb-base",
```
Comment (holden-caulfield) [inline server/.eslintrc.json]: I'm not sure about using `airbnb-base` as an eslint configuration. Specially if we are using prettier. I'm afraid both will clash heavily. I would go for a much simpler eslint config (such as the one that comes with CRA) that does absolutely nothing about formatting, and leave the code style rules to prettier

File: server/.eslintrc.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
...
+    "env": {
+      "mocha": true
```
Comment (holden-caulfield) [inline server/.eslintrc.json]: Do we need mocha? Can't we just use `jest` for everything?

**Comment by gonrial** [comment]
@holden-caulfield several changes were added to #19 to use eslint only for type checking and prettier only for code formatting

File: server/.eslintrc.json (hunk 1 above)
Comment (gonrial) [reply server/.eslintrc.json]: Added to #19

File: server/.eslintrc.json (hunk 2 above)
Comment (gonrial) [reply server/.eslintrc.json]: Added to #20

**Comment by rodrigouroz** [comment]
@holden-caulfield can we let this one go and focus on the other ones?

**Pull request merged**